### PR TITLE
Issues/issue 675 support db 23c

### PR DESCRIPTION
**Pull request by rallen2010**
Fixes Issues #676 and #675  which prevent operation with Oracle database 23c (23.3.0)
- fixes a compile error in flow_log_admin that was allowed in 19c!
- workaround a SQL execution error in 23.3.0 affecting flow_engine and flow_engine_util when getting lane information from diagrams without lanes, which caused Flows for APEX to fail.

The fix for #676 re-touches the file changed in #681 - so I have created a combo patch file that includes fixes 675, 676 and 681.  All three are required to support 23.1.
Runs regression cleanly on 19c and 23c (23.3.0).

I've also made some changes to some regression tests, test apps, and test diagrams / models so that it should be much easier to install them on new test environments, including automated testing - by cleaning up and parameterising app_ids and usernames embedded into test scripts, apps and models.

### CODE REVIEW

File: test/plsql/test_constants.pkb
Comment (commi235) [inline test/plsql/test_constants.pkb:1]: This should be .pks file as it's a package specification.

File: test/testing_readme.md [hunk 1]
```diff
@@ -16,8 +16,19 @@ Using the github repository, go through the following steps to install the test
...
+   
+4. Install the EMP/DEPT Sample Dataset (use APEX or a script)
+
+5. Install any Test Apps in the `test/apps` folder.   These contain things like APEX Approval Task Definitions.   You will need to install the version of these files that matches the APEX version installed in your test environment.
+   
+6. Using the Configurations Panel > Timers (or otherwise) make sure that your timers are enabled and running every 10 seconds.
+   
+7. Using the Configurations Panel > Engine.   Make sure that the default parameters use a valid AppID, App Page and Default User ID on the system undertest.
+   
+8. Edit the Test Constants file (test_constants.pkb) to set any appropriate User IDs and APEX APP IDs for the testing environment.
```
Comment (commi235) [inline test/testing_readme.md:28]: update file name once file is renamed

**Review by commi235 — CHANGES_REQUESTED** [review]
added missing slash and new line to a file.
However that file should also be named differently and references updated.

File: test/plsql/test_constants.pkb
Comment (rallen2010) [reply test/plsql/test_constants.pkb:1]: Oops.   Good catch.   Thanks.

Comment (rallen2010) [reply test/plsql/test_constants.pkb:1]: fixed in commit below

**Comment by rallen2010** [comment]
Sorry, late change - just added test_constants.pks to the test install script as well...

**Review by ralfmueller — APPROVED** [review]

**Review by commi235 — APPROVED** [review]

**Pull request merged**